controllers/pipelines/runschedule_workflow_factory_test.go

```diff
@@ -0,0 +1,46 @@
+//go:build unit
+// +build unit
+
+package pipelines
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sky-uk/kfp-operator/apis"
+	pipelinesv1 "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha5"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var _ = Context("runConfigurationNameForRunSchedule", func() {
+	Specify("returns the name of the owner if set", func() {
+		runSchedule := pipelinesv1.RunSchedule{}
+		runConfiguration := pipelinesv1.RandomRunConfiguration()
+
+		runSchedule.OwnerReferences = []metav1.OwnerReference{{
+			Controller: pointer.Bool(true),
+			APIVersion: "pipelines.kubeflow.org/v1alpha5",
```

Comment:
Can we do something better here to avoid having to update this string when we change version?